125262 – Unify delete draft modal prompts and update tests

6StringCode opened the pull request:
## Summary
Consolidated draft deletion modal titles and content into DELETE_DRAFT_CONFIRM_HEADER and DELETE_DRAFT_CONFIRM_CONTENT in constants. Updated DeleteDraftModal and related unit tests to use the new unified prompt keys for consistency.

Why:
The Modal was showing different messages based on the draft's saved state. UX decided to consolidate and unify all delete draft interactions into one consistent message for all scenarios (saved drafts, unsaved drafts, reply drafts, etc.) instead of showing different messages based on draft state.

Impact:

No test changes required (unit and E2E tests already expect the unified message)
No functional changes to user experience
Cleaner, more maintainable code


- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Related issue(s)

- [_Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#0000](https://github.com/department-of-veterans-affairs/va.gov-team/issues/125262)

## Testing done

- No new functionality to test, but updated `DeleteDraft.unit.spec.jsx` to reflect new consistent content.

## Screenshots 

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        | <img width="329" height="707" alt="Screenshot 2025-12-29 at 11 51 29 AM" src="https://github.com/user-attachments/assets/029de0e4-298a-4248-986e-09bda3bad945" /> |
| Desktop | <img width="703" height="814" alt="Screenshot 2025-12-29 at 1 55 01 PM" src="https://github.com/user-attachments/assets/22aee660-46f0-49ad-9aea-4e2ff93b3acc" /> | <img width="1261" height="1113" alt="Screenshot 2025-12-29 at 11 50 53 AM" src="https://github.com/user-attachments/assets/262eb91e-0d18-491a-b3e1-c36b7a93a63e" /> |

## What areas of the site does it impact?
Secure Messaging

## Acceptance criteria

- [x] Make the alert have consistent content.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added

### Error Handling

- [x] Browser console contains no warnings or errors.

6StringCode committed:
Unify delete draft modal prompts and update tests

Consolidated draft deletion modal titles and content into DELETE_DRAFT_CONFIRM_HEADER and DELETE_DRAFT_CONFIRM_CONTENT in constants. Updated DeleteDraftModal and related unit tests to use the new unified prompt keys for consistency.

6StringCode commented on src/applications/mhv-secure-messaging/components/Modals/DeleteDraftModal.jsx:
```diff
@@ -5,40 +5,30 @@ import { VaModal } from '@department-of-veterans-affairs/component-library/dist/
-          ? Prompts.Draft.DELETE_NEW_DRAFT_TITLE
-          : Prompts.Draft.DELETE_DRAFT_CONFIRM
-      }
+      data-dd-action-name={`${Prompts.Draft.DELETE_DRAFT_CONFIRM_HEADER} Modal${
```
@alex-morgun-1 How can I check this?

alex-morgun-1 replied:
in Datadog RUM

alex-morgun-1 replied:
but I don't think this leading space should impact anything

6StringCode commented on src/applications/mhv-secure-messaging/components/Modals/DeleteDraftModal.jsx:
```diff
@@ -5,40 +5,30 @@ import { VaModal } from '@department-of-veterans-affairs/component-library/dist/
 
 const DeleteDraftModal = props => {
-  const { unsavedDraft, draftSequence } = props;
+  const { draftSequence } = props;
```
Looks like I forgot to remove it from propTypes at the bottom. I'll remove it from the unit test specs as well. Thanks for catching!

6StringCode committed:
Remove unused unsavedDraft prop from DeleteDraftModal

The unsavedDraft prop was removed from DeleteDraftModal and related tests as it is no longer used. This cleans up the component's prop types and test cases.

6StringCode committed:
Remove unused unsavedDraft prop from DeleteDraftModal

The unsavedDraft prop is no longer passed to DeleteDraftModal in DeleteDraft.jsx, likely because it is unused or redundant.
